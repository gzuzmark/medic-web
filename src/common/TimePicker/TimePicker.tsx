import * as moment from 'moment';
import * as React from 'react';
import {default as FilterList, IListItem} from "../FilterList/FilterList";

interface IPropTimePicker {
    defaultText: string;
    from: Date;
    name: string;
    onChange: (item: IListItem) => void;
    step?: number;
    to: Date;
}

const listTimes = (from: Date, to: Date, step = 15): IListItem[] => {
    const times = [] as IListItem[];
    let startTime = moment(from);
    const finishTime = moment(to);
    startTime.set('seconds', 0);
    finishTime.set('seconds', 59);
    while (startTime <= finishTime) {
        times.push({
            id: startTime.toISOString(),
            name: startTime.format('hh:mm a')
        });
        startTime = startTime.add(step, 'minutes');
    }
    return times
};

const TimePicker: React.StatelessComponent<IPropTimePicker> = (props) => {
    return (
        <FilterList
            onChange={props.onChange}
            list={listTimes(props.from, props.to, props.step)}
            defaultText={props.defaultText}
            name={props.name} />
    );
};

TimePicker.defaultProps = {
    step: 15,
};


export default TimePicker;


