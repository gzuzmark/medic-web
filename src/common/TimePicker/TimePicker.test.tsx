import { mount } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import TimePicker from "./TimePicker";


describe('TimePicker Test',() => {
    let props: any;
    let mountedTimePicker: any;
    const getComponent = () => {
        if (!mountedTimePicker) {
            mountedTimePicker = mount(
                <TimePicker {...props} />
            );
        }
        return mountedTimePicker;
    };

    beforeEach(() => {
        const initDate = new Date(2018, 5, 4, 10, 0, 0);
        const endDate = new Date(2018, 5, 4, 12, 0, 0);
        props = {
            defaultText: '11:00',
            from: initDate,
            name: 'Fecha',
            onChange: void(0),
            step: 15,
            to: endDate
        };
        mountedTimePicker = undefined;
    });

    it("render: render TimePicker with two hours", () => {
        const component = getComponent();
        component.find(".FilterList-field").first().simulate('click', "");
        component.update();
        // 10:00, 10:15, 10:30, 10:45, 11:00, 11:15, 11:30, 11:45, 12:00
        expect(component.find("li").length).toBe(9)

    });

    it("render: render TimePicker with two hours every thirty minutes", () => {
        const newProps = {...props};
        newProps.step = 30;
        props = newProps;
        const component = getComponent();
        component.find(".FilterList-field").first().simulate('click', "");
        component.update();
        // 10:00, 10:30, 11:00, 11:30, 12:00
        expect(component.find("li").length).toBe(5)

    });

    it("render: render TimePicker with bad hours", () => {
        const initDate = new Date(2018, 5, 4, 13, 0, 0);
        const endDate = new Date(2018, 5, 4, 10, 0, 0);
        const newProps = {...props};
        newProps.from = initDate;
        newProps.to = endDate;
        props = newProps;
        const component = getComponent();
        component.find(".FilterList-field").first().simulate('click', "");
        component.update();
        expect(component.find("li").length).toBe(0)
    });
});
