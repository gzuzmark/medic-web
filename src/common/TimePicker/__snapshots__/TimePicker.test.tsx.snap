// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker Test render: render TimePicker 1 day 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimePicker
    defaultText="11:00"
    from={"2018-02-20T00:00:00.000Z"}
    name="Inicio"
    onChange={undefined}
    step={15}
    to={"2018-02-21T00:00:00.000Z"}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "defaultText": "11:00",
      "enableClearSearch": false,
      "list": Array [
        Object {
          "id": "2018-02-20T00:00:00.000Z",
          "name": "19:00",
        },
        Object {
          "id": "2018-02-20T00:15:00.000Z",
          "name": "19:15",
        },
        Object {
          "id": "2018-02-20T00:30:00.000Z",
          "name": "19:30",
        },
        Object {
          "id": "2018-02-20T00:45:00.000Z",
          "name": "19:45",
        },
        Object {
          "id": "2018-02-20T01:00:00.000Z",
          "name": "20:00",
        },
        Object {
          "id": "2018-02-20T01:15:00.000Z",
          "name": "20:15",
        },
        Object {
          "id": "2018-02-20T01:30:00.000Z",
          "name": "20:30",
        },
        Object {
          "id": "2018-02-20T01:45:00.000Z",
          "name": "20:45",
        },
        Object {
          "id": "2018-02-20T02:00:00.000Z",
          "name": "21:00",
        },
        Object {
          "id": "2018-02-20T02:15:00.000Z",
          "name": "21:15",
        },
        Object {
          "id": "2018-02-20T02:30:00.000Z",
          "name": "21:30",
        },
        Object {
          "id": "2018-02-20T02:45:00.000Z",
          "name": "21:45",
        },
        Object {
          "id": "2018-02-20T03:00:00.000Z",
          "name": "22:00",
        },
        Object {
          "id": "2018-02-20T03:15:00.000Z",
          "name": "22:15",
        },
        Object {
          "id": "2018-02-20T03:30:00.000Z",
          "name": "22:30",
        },
        Object {
          "id": "2018-02-20T03:45:00.000Z",
          "name": "22:45",
        },
        Object {
          "id": "2018-02-20T04:00:00.000Z",
          "name": "23:00",
        },
        Object {
          "id": "2018-02-20T04:15:00.000Z",
          "name": "23:15",
        },
        Object {
          "id": "2018-02-20T04:30:00.000Z",
          "name": "23:30",
        },
        Object {
          "id": "2018-02-20T04:45:00.000Z",
          "name": "23:45",
        },
        Object {
          "id": "2018-02-20T05:00:00.000Z",
          "name": "00:00",
        },
        Object {
          "id": "2018-02-20T05:15:00.000Z",
          "name": "00:15",
        },
        Object {
          "id": "2018-02-20T05:30:00.000Z",
          "name": "00:30",
        },
        Object {
          "id": "2018-02-20T05:45:00.000Z",
          "name": "00:45",
        },
        Object {
          "id": "2018-02-20T06:00:00.000Z",
          "name": "01:00",
        },
        Object {
          "id": "2018-02-20T06:15:00.000Z",
          "name": "01:15",
        },
        Object {
          "id": "2018-02-20T06:30:00.000Z",
          "name": "01:30",
        },
        Object {
          "id": "2018-02-20T06:45:00.000Z",
          "name": "01:45",
        },
        Object {
          "id": "2018-02-20T07:00:00.000Z",
          "name": "02:00",
        },
        Object {
          "id": "2018-02-20T07:15:00.000Z",
          "name": "02:15",
        },
        Object {
          "id": "2018-02-20T07:30:00.000Z",
          "name": "02:30",
        },
        Object {
          "id": "2018-02-20T07:45:00.000Z",
          "name": "02:45",
        },
        Object {
          "id": "2018-02-20T08:00:00.000Z",
          "name": "03:00",
        },
        Object {
          "id": "2018-02-20T08:15:00.000Z",
          "name": "03:15",
        },
        Object {
          "id": "2018-02-20T08:30:00.000Z",
          "name": "03:30",
        },
        Object {
          "id": "2018-02-20T08:45:00.000Z",
          "name": "03:45",
        },
        Object {
          "id": "2018-02-20T09:00:00.000Z",
          "name": "04:00",
        },
        Object {
          "id": "2018-02-20T09:15:00.000Z",
          "name": "04:15",
        },
        Object {
          "id": "2018-02-20T09:30:00.000Z",
          "name": "04:30",
        },
        Object {
          "id": "2018-02-20T09:45:00.000Z",
          "name": "04:45",
        },
        Object {
          "id": "2018-02-20T10:00:00.000Z",
          "name": "05:00",
        },
        Object {
          "id": "2018-02-20T10:15:00.000Z",
          "name": "05:15",
        },
        Object {
          "id": "2018-02-20T10:30:00.000Z",
          "name": "05:30",
        },
        Object {
          "id": "2018-02-20T10:45:00.000Z",
          "name": "05:45",
        },
        Object {
          "id": "2018-02-20T11:00:00.000Z",
          "name": "06:00",
        },
        Object {
          "id": "2018-02-20T11:15:00.000Z",
          "name": "06:15",
        },
        Object {
          "id": "2018-02-20T11:30:00.000Z",
          "name": "06:30",
        },
        Object {
          "id": "2018-02-20T11:45:00.000Z",
          "name": "06:45",
        },
        Object {
          "id": "2018-02-20T12:00:00.000Z",
          "name": "07:00",
        },
        Object {
          "id": "2018-02-20T12:15:00.000Z",
          "name": "07:15",
        },
        Object {
          "id": "2018-02-20T12:30:00.000Z",
          "name": "07:30",
        },
        Object {
          "id": "2018-02-20T12:45:00.000Z",
          "name": "07:45",
        },
        Object {
          "id": "2018-02-20T13:00:00.000Z",
          "name": "08:00",
        },
        Object {
          "id": "2018-02-20T13:15:00.000Z",
          "name": "08:15",
        },
        Object {
          "id": "2018-02-20T13:30:00.000Z",
          "name": "08:30",
        },
        Object {
          "id": "2018-02-20T13:45:00.000Z",
          "name": "08:45",
        },
        Object {
          "id": "2018-02-20T14:00:00.000Z",
          "name": "09:00",
        },
        Object {
          "id": "2018-02-20T14:15:00.000Z",
          "name": "09:15",
        },
        Object {
          "id": "2018-02-20T14:30:00.000Z",
          "name": "09:30",
        },
        Object {
          "id": "2018-02-20T14:45:00.000Z",
          "name": "09:45",
        },
        Object {
          "id": "2018-02-20T15:00:00.000Z",
          "name": "10:00",
        },
        Object {
          "id": "2018-02-20T15:15:00.000Z",
          "name": "10:15",
        },
        Object {
          "id": "2018-02-20T15:30:00.000Z",
          "name": "10:30",
        },
        Object {
          "id": "2018-02-20T15:45:00.000Z",
          "name": "10:45",
        },
        Object {
          "id": "2018-02-20T16:00:00.000Z",
          "name": "11:00",
        },
        Object {
          "id": "2018-02-20T16:15:00.000Z",
          "name": "11:15",
        },
        Object {
          "id": "2018-02-20T16:30:00.000Z",
          "name": "11:30",
        },
        Object {
          "id": "2018-02-20T16:45:00.000Z",
          "name": "11:45",
        },
        Object {
          "id": "2018-02-20T17:00:00.000Z",
          "name": "12:00",
        },
        Object {
          "id": "2018-02-20T17:15:00.000Z",
          "name": "12:15",
        },
        Object {
          "id": "2018-02-20T17:30:00.000Z",
          "name": "12:30",
        },
        Object {
          "id": "2018-02-20T17:45:00.000Z",
          "name": "12:45",
        },
        Object {
          "id": "2018-02-20T18:00:00.000Z",
          "name": "13:00",
        },
        Object {
          "id": "2018-02-20T18:15:00.000Z",
          "name": "13:15",
        },
        Object {
          "id": "2018-02-20T18:30:00.000Z",
          "name": "13:30",
        },
        Object {
          "id": "2018-02-20T18:45:00.000Z",
          "name": "13:45",
        },
        Object {
          "id": "2018-02-20T19:00:00.000Z",
          "name": "14:00",
        },
        Object {
          "id": "2018-02-20T19:15:00.000Z",
          "name": "14:15",
        },
        Object {
          "id": "2018-02-20T19:30:00.000Z",
          "name": "14:30",
        },
        Object {
          "id": "2018-02-20T19:45:00.000Z",
          "name": "14:45",
        },
        Object {
          "id": "2018-02-20T20:00:00.000Z",
          "name": "15:00",
        },
        Object {
          "id": "2018-02-20T20:15:00.000Z",
          "name": "15:15",
        },
        Object {
          "id": "2018-02-20T20:30:00.000Z",
          "name": "15:30",
        },
        Object {
          "id": "2018-02-20T20:45:00.000Z",
          "name": "15:45",
        },
        Object {
          "id": "2018-02-20T21:00:00.000Z",
          "name": "16:00",
        },
        Object {
          "id": "2018-02-20T21:15:00.000Z",
          "name": "16:15",
        },
        Object {
          "id": "2018-02-20T21:30:00.000Z",
          "name": "16:30",
        },
        Object {
          "id": "2018-02-20T21:45:00.000Z",
          "name": "16:45",
        },
        Object {
          "id": "2018-02-20T22:00:00.000Z",
          "name": "17:00",
        },
        Object {
          "id": "2018-02-20T22:15:00.000Z",
          "name": "17:15",
        },
        Object {
          "id": "2018-02-20T22:30:00.000Z",
          "name": "17:30",
        },
        Object {
          "id": "2018-02-20T22:45:00.000Z",
          "name": "17:45",
        },
        Object {
          "id": "2018-02-20T23:00:00.000Z",
          "name": "18:00",
        },
        Object {
          "id": "2018-02-20T23:15:00.000Z",
          "name": "18:15",
        },
        Object {
          "id": "2018-02-20T23:30:00.000Z",
          "name": "18:30",
        },
        Object {
          "id": "2018-02-20T23:45:00.000Z",
          "name": "18:45",
        },
        Object {
          "id": "2018-02-21T00:00:00.000Z",
          "name": "19:00",
        },
      ],
      "name": "Inicio",
      "onChange": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultText": "11:00",
        "enableClearSearch": false,
        "list": Array [
          Object {
            "id": "2018-02-20T00:00:00.000Z",
            "name": "19:00",
          },
          Object {
            "id": "2018-02-20T00:15:00.000Z",
            "name": "19:15",
          },
          Object {
            "id": "2018-02-20T00:30:00.000Z",
            "name": "19:30",
          },
          Object {
            "id": "2018-02-20T00:45:00.000Z",
            "name": "19:45",
          },
          Object {
            "id": "2018-02-20T01:00:00.000Z",
            "name": "20:00",
          },
          Object {
            "id": "2018-02-20T01:15:00.000Z",
            "name": "20:15",
          },
          Object {
            "id": "2018-02-20T01:30:00.000Z",
            "name": "20:30",
          },
          Object {
            "id": "2018-02-20T01:45:00.000Z",
            "name": "20:45",
          },
          Object {
            "id": "2018-02-20T02:00:00.000Z",
            "name": "21:00",
          },
          Object {
            "id": "2018-02-20T02:15:00.000Z",
            "name": "21:15",
          },
          Object {
            "id": "2018-02-20T02:30:00.000Z",
            "name": "21:30",
          },
          Object {
            "id": "2018-02-20T02:45:00.000Z",
            "name": "21:45",
          },
          Object {
            "id": "2018-02-20T03:00:00.000Z",
            "name": "22:00",
          },
          Object {
            "id": "2018-02-20T03:15:00.000Z",
            "name": "22:15",
          },
          Object {
            "id": "2018-02-20T03:30:00.000Z",
            "name": "22:30",
          },
          Object {
            "id": "2018-02-20T03:45:00.000Z",
            "name": "22:45",
          },
          Object {
            "id": "2018-02-20T04:00:00.000Z",
            "name": "23:00",
          },
          Object {
            "id": "2018-02-20T04:15:00.000Z",
            "name": "23:15",
          },
          Object {
            "id": "2018-02-20T04:30:00.000Z",
            "name": "23:30",
          },
          Object {
            "id": "2018-02-20T04:45:00.000Z",
            "name": "23:45",
          },
          Object {
            "id": "2018-02-20T05:00:00.000Z",
            "name": "00:00",
          },
          Object {
            "id": "2018-02-20T05:15:00.000Z",
            "name": "00:15",
          },
          Object {
            "id": "2018-02-20T05:30:00.000Z",
            "name": "00:30",
          },
          Object {
            "id": "2018-02-20T05:45:00.000Z",
            "name": "00:45",
          },
          Object {
            "id": "2018-02-20T06:00:00.000Z",
            "name": "01:00",
          },
          Object {
            "id": "2018-02-20T06:15:00.000Z",
            "name": "01:15",
          },
          Object {
            "id": "2018-02-20T06:30:00.000Z",
            "name": "01:30",
          },
          Object {
            "id": "2018-02-20T06:45:00.000Z",
            "name": "01:45",
          },
          Object {
            "id": "2018-02-20T07:00:00.000Z",
            "name": "02:00",
          },
          Object {
            "id": "2018-02-20T07:15:00.000Z",
            "name": "02:15",
          },
          Object {
            "id": "2018-02-20T07:30:00.000Z",
            "name": "02:30",
          },
          Object {
            "id": "2018-02-20T07:45:00.000Z",
            "name": "02:45",
          },
          Object {
            "id": "2018-02-20T08:00:00.000Z",
            "name": "03:00",
          },
          Object {
            "id": "2018-02-20T08:15:00.000Z",
            "name": "03:15",
          },
          Object {
            "id": "2018-02-20T08:30:00.000Z",
            "name": "03:30",
          },
          Object {
            "id": "2018-02-20T08:45:00.000Z",
            "name": "03:45",
          },
          Object {
            "id": "2018-02-20T09:00:00.000Z",
            "name": "04:00",
          },
          Object {
            "id": "2018-02-20T09:15:00.000Z",
            "name": "04:15",
          },
          Object {
            "id": "2018-02-20T09:30:00.000Z",
            "name": "04:30",
          },
          Object {
            "id": "2018-02-20T09:45:00.000Z",
            "name": "04:45",
          },
          Object {
            "id": "2018-02-20T10:00:00.000Z",
            "name": "05:00",
          },
          Object {
            "id": "2018-02-20T10:15:00.000Z",
            "name": "05:15",
          },
          Object {
            "id": "2018-02-20T10:30:00.000Z",
            "name": "05:30",
          },
          Object {
            "id": "2018-02-20T10:45:00.000Z",
            "name": "05:45",
          },
          Object {
            "id": "2018-02-20T11:00:00.000Z",
            "name": "06:00",
          },
          Object {
            "id": "2018-02-20T11:15:00.000Z",
            "name": "06:15",
          },
          Object {
            "id": "2018-02-20T11:30:00.000Z",
            "name": "06:30",
          },
          Object {
            "id": "2018-02-20T11:45:00.000Z",
            "name": "06:45",
          },
          Object {
            "id": "2018-02-20T12:00:00.000Z",
            "name": "07:00",
          },
          Object {
            "id": "2018-02-20T12:15:00.000Z",
            "name": "07:15",
          },
          Object {
            "id": "2018-02-20T12:30:00.000Z",
            "name": "07:30",
          },
          Object {
            "id": "2018-02-20T12:45:00.000Z",
            "name": "07:45",
          },
          Object {
            "id": "2018-02-20T13:00:00.000Z",
            "name": "08:00",
          },
          Object {
            "id": "2018-02-20T13:15:00.000Z",
            "name": "08:15",
          },
          Object {
            "id": "2018-02-20T13:30:00.000Z",
            "name": "08:30",
          },
          Object {
            "id": "2018-02-20T13:45:00.000Z",
            "name": "08:45",
          },
          Object {
            "id": "2018-02-20T14:00:00.000Z",
            "name": "09:00",
          },
          Object {
            "id": "2018-02-20T14:15:00.000Z",
            "name": "09:15",
          },
          Object {
            "id": "2018-02-20T14:30:00.000Z",
            "name": "09:30",
          },
          Object {
            "id": "2018-02-20T14:45:00.000Z",
            "name": "09:45",
          },
          Object {
            "id": "2018-02-20T15:00:00.000Z",
            "name": "10:00",
          },
          Object {
            "id": "2018-02-20T15:15:00.000Z",
            "name": "10:15",
          },
          Object {
            "id": "2018-02-20T15:30:00.000Z",
            "name": "10:30",
          },
          Object {
            "id": "2018-02-20T15:45:00.000Z",
            "name": "10:45",
          },
          Object {
            "id": "2018-02-20T16:00:00.000Z",
            "name": "11:00",
          },
          Object {
            "id": "2018-02-20T16:15:00.000Z",
            "name": "11:15",
          },
          Object {
            "id": "2018-02-20T16:30:00.000Z",
            "name": "11:30",
          },
          Object {
            "id": "2018-02-20T16:45:00.000Z",
            "name": "11:45",
          },
          Object {
            "id": "2018-02-20T17:00:00.000Z",
            "name": "12:00",
          },
          Object {
            "id": "2018-02-20T17:15:00.000Z",
            "name": "12:15",
          },
          Object {
            "id": "2018-02-20T17:30:00.000Z",
            "name": "12:30",
          },
          Object {
            "id": "2018-02-20T17:45:00.000Z",
            "name": "12:45",
          },
          Object {
            "id": "2018-02-20T18:00:00.000Z",
            "name": "13:00",
          },
          Object {
            "id": "2018-02-20T18:15:00.000Z",
            "name": "13:15",
          },
          Object {
            "id": "2018-02-20T18:30:00.000Z",
            "name": "13:30",
          },
          Object {
            "id": "2018-02-20T18:45:00.000Z",
            "name": "13:45",
          },
          Object {
            "id": "2018-02-20T19:00:00.000Z",
            "name": "14:00",
          },
          Object {
            "id": "2018-02-20T19:15:00.000Z",
            "name": "14:15",
          },
          Object {
            "id": "2018-02-20T19:30:00.000Z",
            "name": "14:30",
          },
          Object {
            "id": "2018-02-20T19:45:00.000Z",
            "name": "14:45",
          },
          Object {
            "id": "2018-02-20T20:00:00.000Z",
            "name": "15:00",
          },
          Object {
            "id": "2018-02-20T20:15:00.000Z",
            "name": "15:15",
          },
          Object {
            "id": "2018-02-20T20:30:00.000Z",
            "name": "15:30",
          },
          Object {
            "id": "2018-02-20T20:45:00.000Z",
            "name": "15:45",
          },
          Object {
            "id": "2018-02-20T21:00:00.000Z",
            "name": "16:00",
          },
          Object {
            "id": "2018-02-20T21:15:00.000Z",
            "name": "16:15",
          },
          Object {
            "id": "2018-02-20T21:30:00.000Z",
            "name": "16:30",
          },
          Object {
            "id": "2018-02-20T21:45:00.000Z",
            "name": "16:45",
          },
          Object {
            "id": "2018-02-20T22:00:00.000Z",
            "name": "17:00",
          },
          Object {
            "id": "2018-02-20T22:15:00.000Z",
            "name": "17:15",
          },
          Object {
            "id": "2018-02-20T22:30:00.000Z",
            "name": "17:30",
          },
          Object {
            "id": "2018-02-20T22:45:00.000Z",
            "name": "17:45",
          },
          Object {
            "id": "2018-02-20T23:00:00.000Z",
            "name": "18:00",
          },
          Object {
            "id": "2018-02-20T23:15:00.000Z",
            "name": "18:15",
          },
          Object {
            "id": "2018-02-20T23:30:00.000Z",
            "name": "18:30",
          },
          Object {
            "id": "2018-02-20T23:45:00.000Z",
            "name": "18:45",
          },
          Object {
            "id": "2018-02-21T00:00:00.000Z",
            "name": "19:00",
          },
        ],
        "name": "Inicio",
        "onChange": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TimePicker Test render: render TimePicker default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimePicker
    defaultText="11:00"
    from={"2018-02-20T00:00:00.000Z"}
    name="Inicio"
    onChange={undefined}
    step={15}
    to={"2018-02-20T00:00:00.000Z"}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "defaultText": "11:00",
      "enableClearSearch": false,
      "list": Array [
        Object {
          "id": "2018-02-20T00:00:00.000Z",
          "name": "19:00",
        },
      ],
      "name": "Inicio",
      "onChange": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultText": "11:00",
        "enableClearSearch": false,
        "list": Array [
          Object {
            "id": "2018-02-20T00:00:00.000Z",
            "name": "19:00",
          },
        ],
        "name": "Inicio",
        "onChange": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
